tests/foreman/maintain/test_advanced.py

```diff
@@ -0,0 +1,434 @@
+
+    :steps:
+        1. Change password for user to any string
+        2. Run advanced procedure run hammer-setup with wrong_password.
+        3. Verify wrong_password isn't updated in MAINTAIN_HAMMER_YML
+        4. Run advanced procedure run hammer-setup with changed password.
+        5. Verify changed password is updated in MAINTAIN_HAMMER_YML
+        6. Update admin password back to default
+        7. Verify default password is updated in MAINTAIN_HAMMER_YML
+
+    :expectedresults:
+        1. Run hammer setup with wrong password, it should fail and
+           password shouldn't be updated in MAINTAIN_HAMMER_YML
+        2. Run hammer setup with changed password, it should pass and
+           password should be updated in MAINTAIN_HAMMER_YML
+
+    :BZ: 1830355
+    """
+    default_admin_pass = settings.server.admin_password
+    result = sat_maintain.execute(
+        f'hammer -u admin -p {default_admin_pass} user update --login admin --password admin'
+    )
+    assert result.status == 0
+
+    # try with incorrect password
+    result = sat_maintain.cli.Advanced.run_hammer_setup(env_var='echo "wrong_password" | ')
+    assert 'Incorrect credential for admin user' in result.stdout
+    assert result.status == 1
+
+    # Verify wrong_password isn't updated in MAINTAIN_HAMMER_YML
+    result = sat_maintain.execute(f"grep -i ':password: wrong_password' {MAINTAIN_HAMMER_YML}")
+    assert result.status != 0
+    assert 'wrong_password' not in result.stdout
+
+    # try with correct password
+    result = sat_maintain.cli.Advanced.run_hammer_setup(env_var='echo "admin" | ')
+    assert result.status == 0
+
+    # Verify new password updated in MAINTAIN_HAMMER_YML
+    result = sat_maintain.execute(f"grep -i ':password: admin' {MAINTAIN_HAMMER_YML}")
+    assert result.status == 0
+    assert 'admin' in result.stdout
+
+    @request.addfinalizer
+    def _finalize():
+        result = sat_maintain.execute(
+            f'hammer -u admin -p admin user update --login admin --password {default_admin_pass}'
+        )
+        assert result.status == 0
+        # Make default admin creds available in MAINTAIN_HAMMER_YML
+        result = sat_maintain.cli.Advanced.run_hammer_setup()
+        assert result.status == 0
+        # Make sure default password available in MAINTAIN_HAMMER_YML
+        result = sat_maintain.execute(
+            f"grep -i ':password: {default_admin_pass}' {MAINTAIN_HAMMER_YML}"
+        )
+        assert result.status == 0
+        assert default_admin_pass in result.stdout
+
+
+@pytest.mark.stubbed
+def test_positive_satellite_maintain_packages_update(sat_maintain):
+    """Packages update using advanced procedure run
+
+    :id: d56523d7-7042-40e1-a96e-db8f33b960e5
+
+    :steps:
+        1. Run satellite-maintain advanced procedure run packages-update
+
+    :expectedresults: packages should update.
+    """
+    result = sat_maintain.cli.Advanced.run_packages_update()
```

Comment:
Is this a synchronous operation so that it waits for the command to run till the end and then we check if `FAIL` in there.

Reply:
Can you please reply before self resolving the conversation :)